contributors/design-proposals/bulk_watch.md

```diff
@@ -0,0 +1,328 @@
+# Background
+
+ As part of increasing security of a cluster, we are planning to limit the ability
+of a given Kubelet (in general: node), to be able to read only resources associated
+with it. Those resources, in particular means: secrets, configmaps &
+persistentvolumeclaims. This is needed to avoid situation when compromising node
+de facto means compromising a cluster. For more details & discussions see
+https://github.com/kubernetes/kubernetes/issues/40476.
+ 
+ However, by some extension to this effort, we would like to improve scalability
+of the system, by significantly reducing amount of api calls coming from kubelets.
+As of now, to avoid situation that kubelet is watching all secrets/configmaps/...
+in the system, it is not using watch. Instead of that, it is retrieving indidual
+objects, by sending individual GET requests. However, it is sending those requests
+periodically to enable automatic updates of mounted secrets/configmaps/...
+In large clusters, this is generating huge unnecessary load, as this load in
+principle should be watch-based. We would like to address this together with
+solving the authorization issue.
+
+
+# Proposal
+
+ In this proposal, we're not focusing on how exactly security should be done.
+We're just sketching very high level approach and exact authorization mechanism
+should be discussed separately.
+
+ At the high level, what we would like to achieve is to enable LIST and WATCH
+requests to support more sophisticated filtering (e.g. we would like to be able
+to somehow ask for all Secrets attached to all pods bound to a given node in a
+single LIST or WATCH request). We obviously want to be able to authorize all
+other types of requests (in particular GETs), so that the design has to be
+consistent in all those cases.
```

Comment:
Added a paragraph.

However, i think this isn't specific to the "bulk watch" and we should provide the same mechanism in regular watch.  As a user, I can imagine that I'm watching e.g. a single secret (using regular watch) and in this case also lose access to it. In this case my watch also should be automatically closed. Added a TODO to reflect my concern that it's not bulk-specific.